cmd/manager/main.go

```diff
@@ -106,17 +107,29 @@ func main() {
 		os.Exit(1)
 	}
 
+	if err := monitoringv1.AddToScheme(mgr.GetScheme()); err != nil {
+		log.Error(err, "error registering prometheus monitoring objects")
+		os.Exit(1)
+	}
+
 	// Setup all Controllers
 	if err := controller.AddToManager(mgr); err != nil {
 		log.Error(err, "")
 		os.Exit(1)
 	}
 
-	// Create Service object to expose the metrics port.
-	_, err = metrics.ExposeMetricsPort(ctx, metricsPort)
+	s, _ := operatormetrics.GenerateService(8080, "prom-metrics")
+	sm := operatormetrics.GenerateServiceMonitor(s)
+	err = mgr.GetClient().Create(context.TODO(), s)
+	if err != nil {
+		log.Error(err, "error creating metrics Service")
+	}
+	log.Info("Created Service")
+	err = mgr.GetClient().Create(context.TODO(), sm)
 	if err != nil {
-		log.Info(err.Error())
+		log.Error(err, "error creating metrics ServiceMonitor")
```

Comment:
AFAICS we are not exiting if the service/sm creation fails, I would exit and abort the operator startup if it fails

Reply:
I couldn't decide if it should exit or not, happy to add it

Reply:
I think it makes sense to exit, to avoid having a metric-less operator running

Reply:
@fahlmant I would prefer to not exit if there is problem with registering metrics. Core functionality of the operator is not impacted when unable to export metrics.